update_deps/propagate_updates.ps1

```diff
@@ -1,2 +1,509 @@
+    $body = @(
+        @{
+            op='add'
+            path='/relations/-'
+            value=@{
+                rel='ArtifactLink'
+                url=$pr_artifact_id
+                attributes=@{
+                    name="pull request"
+                }
+            }
+        }
+    )
+    $body = $body | ConvertTo-Json
+    # This PATCH endpoint takes a list of patch objects so $body must be encased in a list
+    $response = Invoke-WebRequest -URI $request_url -UseBasicParsing -Method Patch -Headers $azure_header -Body "[$body]" -ContentType "application/json-patch+json"
+    if(!$response) {
+        Write-Error "Failed to link work item to PR.`nWork item: $work_item_id`nPR: $pr_artifact_id"
+        exit -1
+    }
+}
+
+
+# approve PR for Azure repo
+function approve-pr-azure {
+    param(
+        [string] $pr_url,
+        [string] $creator_id
+    )
+    $request_url = $pr_url + '/reviewers/' + $creator_id + '?api-version=6.0'
+    $body = @{
+        vote=10
+    }
+    $body = $body | ConvertTo-Json
+    $response = Invoke-WebRequest -URI $request_url -UseBasicParsing -Method Put -Headers $azure_header -Body $body -ContentType "application/json"
+    if(!$response) {
+        Write-Error "Failed to approve PR: $pr_url"
+        exit -1
+    }
+}
+
+
+# set PR for Azure repo to merge automatically once build completes
+function set-autocomplete-azure {
+    param(
+        $pr_url,
+        $creator_id
+    )
+    $request_url = $pr_url + '?api-version=6.0'
+    $body =@{
+        autoCompleteSetBy=@{
+            id=$creator_id
+        }
+        completionOptions=@{
+            mergeStrategy="squash"
+        }
+    }
+    $body = $body | ConvertTo-Json
+    $response = Invoke-WebRequest -URI $request_url -UseBasicParsing -Method Patch -Headers $azure_header -Body $body -ContentType "application/json"
+    if(!$response) {
+        Write-Error "Failed to set autocomplete for PR $pr_url"
+        exit -1
+    }
+}
+
+
+# wait until build completes for Azure repo
+function wait-until-complete-azure {
+    param(
+        [string] $pr_url
+    )
+    $status = ""
+    while($true){
+        $response = Invoke-WebRequest -URI $pr_url -UseBasicParsing -Method Get -Headers $azure_header
+        if(!$response) {
+            Write-Error "Failed to get PR: $pr_url"
+            exit -1
+        }
+        $content =  $response.Content | ConvertFrom-Json
+        $status = $content.status
+        $merge_status = $content.mergeStatus
+        if($status -ne "active" -or $merge_status -ne "succeeded") {
+            break
+        }
+        spin 10
+    }
+    if($status -ne "completed") {
+        Write-Host "Problem with pull request: $pr_url"
+        Write-Error $response.Content
+        exit -1
+    }
+}
+
+
+# update dependencies for Azure repo
+function update-repo-azure {
+    param(
+        [string] $repo_name
+    )
+    Write-Host "`nCreating work item"
+    $work_item_id = create-work-item-azure
```

Comment:
Frankly generating work items does not add a lot of value.
The reasoin to attach a work item is to be able to track from a commit to a PR from the PR to a work item. As in: "why did we do the change"?
Or at least that is the imagined reason why we would do it, since it is extremely rare that we trace code changes like that.

If we generate the work item the answer to that question does not exist anymore.

I think it would be better to either:
i) ask the user for the workitem id.     or
ii) allow submodule updates without specifying the work item.

Reply:
In the next iteration, user is asked to provide work item id.